packages/dbml-core/src/parse/Parser.js

```diff
@@ -55,6 +60,31 @@ class Parser {
         rawDatabase = Parser.parseDBMLToJSON(str);
         break;
 
+      case 'dbmlv2': {
+        this.DBMLCompiler.setSource(str);
+
+        const errors = this.DBMLCompiler.parse.errors();
+        if (errors.length > 0) {
+          throw errors.map((error) => ({
+            message: error.diagnostic,
+            location: {
+              start: {
+                line: error.nodeOrToken.startPos.line + 1,
+                column: error.nodeOrToken.startPos.column + 1,
+              },
+              end: {
+                line: error.nodeOrToken.endPos.line + 1,
+                column: error.nodeOrToken.endPos.column + 1,
+              },
+            },
+            code: error.code,
+          }));
+        }
+
+        rawDatabase = this.DBMLCompiler.parse.rawDb();
+        break;
+      }
```

Comment:
There're several problems with our Parser error formats:
* Structure and type not consistent between various SQL parser and DBML
* The error types is not clearly defined somewhere

Improvement: should refactor error system in the future